### PR TITLE
Terraform upgrade guide

### PR DESCRIPTION
**Pull request by nie7321**
## Overview

Adds a Terraform upgrade guide for AS.

## Checklist

- [x] Proof-reading
- [x] Any diagrams/figures have their source files included

### CODE REVIEW

File: iac/tf-upgrading.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# Upgrading Terraform
+Prior to January 2020, all IaC was written in Terraform v0.10.x. One selected version of the `terraform` command was available on the Jenkins servers.
+
+The v0.10 series has reached the end of its useful life: the AWS provider will [no longer be updated](https://www.hashicorp.com/blog/deprecating-terraform-0-11-support-in-terraform-providers/) for Terraform 0.11 and older. Without active support on the AWS provider, new AWS features will be unavailable to us. As time goes on, AWS API changes may leave existing IaC modules broken & unable to build.
+
+## Strategy
+A utility called [`tfenv`](https://github.com/tfutils/tfenv) is available on the Jenkins servers. This is similar to Python's `virtual_env`, Node's `nvm`, or Ruby's `rbenv`. It allows you to select a specific version of Terraform on a per-project basis.
```
Comment (jwalsh2) [inline iac/tf-upgrading.md]: Very cool -- was not aware of `tfenv`

File: iac/tf-upgrading.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+This utility will allow us to run multiple versions of Terraform on the same Jenkins server. Projects may be upgraded one-by-one, instead of needing all jobs on one Jenkins server to upgrade to Terraform v0.12 in unison.
+
```
Comment (jwalsh2) [inline iac/tf-upgrading.md]: I had to provider lock some versions before I left on some of the v.10 TF code for the aws modules to keep from breaking some syntax... be aware.

Comment (nie7321) [reply iac/tf-upgrading.md]: Oh, good to know, I'll add a note about looking for that to the docs!

I _think_ dependabot will monitor that stuff too, if we set it up for the IaC repos.

**Comment by nie7321** [comment]
Your approval always matters to us @jwalsh2 ❤️

File: iac/tf-upgrading.md (hunk 2 above)
Comment (nie7321) [reply iac/tf-upgrading.md]: All set. Thanks for pointing that out!

**Pull request merged**